### ISSUE TITLE
Different inferred type `t = tuple(); list(t)` and `list(tuple())`

### ISSUE THREAD

**Issue by InSyncWithFoo** [microsoft/pyright#7998]
### Discussed in https://github.com/microsoft/pyright/discussions/7996

<div type='discussions-op-text'>

<sup>Originally posted by **InSyncWithFoo** May 24, 2024</sup>
```python
def bar():
    t = tuple()
    reveal_type(t)  # tuple[Unknown, ...]
    return list(t)

def qux():
    return list(tuple())
```

```python
reveal_type(bar)    # () -> list[Unknown]
reveal_type(qux)    # () -> list[_T@list]
```

Only `bar` is reported as having a partially unknown return type. Instead of `Unknown`, the inferred type for `qux` has an unresolved type variable. On the other hand, `qux()` is of type `list[Unknown]`. Shouldn't 	`qux`, too, be reported?

</div>

**Comment by erictraut** [comment]
This is included in pyright 1.1.365.
